Services/CO.CDP.Organisation.WebApi/UseCase/UpdateOrganisationUseCase.cs

```diff
@@ -189,12 +196,71 @@ public async Task<bool> Execute((Guid organisationId, UpdateOrganisation updateO
 
         organisation.UpdateSupplierInformation();
 
+        await ResetRejectedStatus(command.updateOrganisation.Type, organisation);
+
         await organisationRepository.SaveAsync(organisation,
             async o => await publisher.Publish(mapper.Map<OrganisationUpdated>(o)));
 
         return await Task.FromResult(true);
     }
 
+    private async Task ResetRejectedStatus(OrganisationUpdateType updateType, Persistence.Organisation organisation)
+    {
+        if (
+            (updateType == OrganisationUpdateType.OrganisationName || updateType == OrganisationUpdateType.OrganisationEmail)
+            && organisation.PendingRoles.Contains(PartyRole.Buyer)
+            && organisation.ReviewedBy != null
+        )
+        {
+            await ResendBuyerApprovalEmail(organisation);
+
+            // TODO: We're "resetting" the review status here but the only way is to wipe the previous review. We should likely refactor this to have a review_status column.
+            organisation.ReviewComment = null;
+            organisation.ReviewedBy = null;
```

Comment:
This is a bit grim. Again, if we had the nice "status" column then we would just reset it here. But instead I'm having to wipe out the previous review in order to reset their status from rejected back to pending approval